pkg/util/steps/condition.go

```diff
@@ -66,9 +70,48 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
+	}
 	return err
 }
 
+func (c conditionStep) enrichConditionTimeoutError() error {
+	funcNameParts := strings.Split(FriendlyName(c.f), ".")
+	funcName := strings.Replace(funcNameParts[len(funcNameParts)-1], "-fm", "", 1)
+
+	message := "timed out polling for a prerequisite cluster condition to resolve"
```

Comment:
You set this string twice - once here, once in `default`. Likely best to only be once.